Development/nmos/connection_api.cpp

```diff
@@ -455,6 +456,24 @@ namespace nmos
                         // find resource again just in case, since waiting releases and reacquires the lock
                         resource = find_resource(resources, id_type);
                     }
+
+                    // "At any time if State of an active Sender becomes active_constraints_violation, the Sender MUST become inactive.
+                    // An inactive Sender in this state MUST NOT allow activations.
+                    // At any time if State of an active Receiver becomes non_compliant_stream, the Receiver SHOULD become inactive.
+                    // An inactive Receiver in this state SHOULD NOT allow activations."
+                    // See https://specs.amwa.tv/is-11/branches/v1.0-dev/docs/Behaviour_-_Server_Side.html#preventing-restrictions-violation
+                    auto streamcompatibility_resource = find_resource(model.streamcompatibility_resources, id_type);
+                    if (model.streamcompatibility_resources.end() != streamcompatibility_resource)
+                    {
+                        auto resource_state = nmos::fields::state(nmos::fields::status(streamcompatibility_resource->data));
+
+                        if (resource_state == nmos::sender_states::active_constraints_violation.name || resource_state == nmos::receiver_states::non_compliant_stream.name)
+                        {
+                            slog::log<slog::severities::error>(gate, SLOG_FLF) << "Rejecting PATCH request for " << id_type << " due to its state: " << resource_state;
```

Comment:
It's a great idea! I don't want to couple IS-11 with any other API implementation and `validate_merged` looks good for this purpose. Decoupled them and added a `master_enable` check.

Reply:
Nice, OK.

If the implementation of this check would be common, you could provide that in the library, based on current lambda in `make_node_implementation_patch_validator`? Not quite sure where it would live, maybe in _nmos/connection_streamcompatibility_validator.{cpp,h}_ (inspired by _nmos/connection_events_activation.{cpph}_).

Reply:
Yes, it's common. Added `make_connection_streamcompatibility_validator` to _nmos/streamcompatibility_validation.{cpp,h}_.